tiledb/bioimg/converters/base.py

```diff
@@ -58,80 +60,64 @@
 from .tiles import iter_tiles, num_tiles
 
 
-class ImageReader(ABC):
-    @abstractmethod
-    def __init__(
-        self,
-        input_path: str,
-        *,
-        logger: Optional[logging.Logger],
-        source_config: Optional[tiledb.Config] = None,
-        source_ctx: Optional[tiledb.Ctx] = None,
-        dest_config: Optional[tiledb.Config] = None,
-        dest_ctx: Optional[tiledb.Config] = None,
-        **kwargs: Any,
-    ):
-        """Initialize this ImageReader"""
+class ImageReader(Protocol):
 
-    def __enter__(self) -> ImageReader:
-        return self
+    def __enter__(self) -> ImageReader: ...
```

Comment:
I imagine you’ll want to make this a `Self`, since that explicitly says “the thing you return here is of the same type as me”. Right now it would be allowable (if strange) to have

```python
reader = ImageReaderA()
with reader as with_block_reader:
  reveal_type(with_block_reader)  # ImageReaderB
```

Reply:
Sure I will use the `TReader` and `TWriter` respectively as the supported version is `pre-3.11`